fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/ColumnRangePredicate.java

```diff
@@ -73,27 +86,89 @@ public static ColumnRangePredicate andRange(
                 }
             }
         }
-        return new ColumnRangePredicate(rangePredicate.columnRef, TreeRangeSet.create(ranges));
+        return new ColumnRangePredicate(rangePredicate.getExpression(), TreeRangeSet.create(ranges));
     }
 
     public static ColumnRangePredicate orRange(
             ColumnRangePredicate rangePredicate, ColumnRangePredicate otherRangePredicate) {
         TreeRangeSet<ConstantOperator> result = TreeRangeSet.create();
         result.addAll(rangePredicate.columnRanges);
         result.addAll(otherRangePredicate.columnRanges);
-        return new ColumnRangePredicate(rangePredicate.getColumnRef(), result);
+        return new ColumnRangePredicate(rangePredicate.getExpression(), result);
     }
 
     public boolean isUnbounded() {
         return columnRanges.asRanges().stream().allMatch(range -> !range.hasUpperBound() && !range.hasLowerBound());
     }
 
+    public List<ColumnRangePredicate> getEquivalentRangePredicates() {
+        if (isCastDate() || isStr2Date()) {
```

Comment:
how to extend to some other functions, like `str_to_date`, `str_to_jodatime` ?

Reply:
1. add more if else
2. use ScalarOperatorVisitor to process different expressions.